MapboxCoreNavigation/RouteController.swift

```diff
@@ -238,7 +238,7 @@ open class RouteController: NSObject, Router {
     @objc public func userIsOnRoute(_ location: CLLocation) -> Bool {
         
         // If the user has arrived, do not continue monitor reroutes, step progress, etc
-        guard !routeProgress.currentLegProgress.userHasArrivedAtWaypoint || (delegate?.router?(self, shouldPreventReroutesWhenArrivingAt: routeProgress.currentLeg.destination) ?? DefaultBehavior.shouldPreventReroutesWhenArrivingAtWaypoint) else {
+        guard !routeProgress.currentLegProgress.userHasArrivedAtWaypoint || !(delegate?.router?(self, shouldPreventReroutesWhenArrivingAt: routeProgress.currentLeg.destination) ?? DefaultBehavior.shouldPreventReroutesWhenArrivingAtWaypoint) else {
```

Comment:
There are a few layers of double negatives in the existing code, made worse by the `guard` statement. If we reexpress the original code (using De Morgan’s laws) as:

```swift
if routeProgress.currentLegProgress.userHasArrivedAtWaypoint
    && !(delegate?.router?(self, shouldPreventReroutesWhenArrivingAt: routeProgress.currentLeg.destination)
         ?? DefaultBehavior.shouldPreventReroutesWhenArrivingAtWaypoint) {
```

then it’s a little clearer that the method says the user is on the route (preventing rerouting) if `userHasArrivedAtWaypoint` is set and the delegate declines to prevent a reroute (that is, allows rerouting to occur). You’re right; that’s incorrect.

This change says the user is on the route when `userHasArrivedAtWaypoint` is set and the delegate says it wants to prevent rerouting.